packages/dds/shared-object-base/src/sharedObject.ts

```diff
@@ -344,21 +344,53 @@ export abstract class SharedObjectCore<TEvent extends ISharedObjectEvents = ISha
 	 */
 	protected abstract onDisconnect();
 
+	//* TODO: Add doc comments for rootMetadata throughout
+
 	/**
 	 * Submits a message by the local client to the runtime.
 	 * @param content - Content of the message
 	 * @param localOpMetadata - The local metadata associated with the message. This is kept locally by the runtime
 	 * and not sent to the server. This will be sent back when this message is received back from the server. This is
 	 * also sent if we are asked to resubmit the message.
 	 */
-	protected submitLocalMessage(content: any, localOpMetadata: unknown = undefined): void {
+	protected submitLocalMessage(
+		content: any,
+		localOpMetadata: unknown,
+		rootMetadata: unknown,
```

Comment:
I would not make this an `unknown` type but make it very specific to be a list of handles in the content. That makes the API simple for DDS authors and the conversion from handle to outbound routes can either be done here in `SharedObject` or in container runtime layer.
The `FluidSerializer`'s encode / serialize methods could return a list of handles along with the serialized content to make it easier to get this list from user content.

Reply:
Yeah it won't be `unknown`.  I just did that for this first pass up getting the plumbing right.  I'm not going to merge this PR until we finalize that type based on the coding of the actual feature as you're describing.  I'm thinking it'll be 

```ts
rootMetadata: {
  newOutboundRoutes: [string, string][]; // [ [ toId, fromId], [toId, fromId], ... ]
}
```

And yes you described the intended change to `FluidSerializer` similar to how I'm picturing it.  Just didn't get to coding it up due to all the other GC concerns that came up as I went along.

Reply:
I updated it to have an actual named type which is an empty interface for now, to avoid confusion.